### DISCUSSION TITLE
Mach number exceeds 1.0. Equations cannot be evaluated.

### DISCUSSION THREAD

**Discussion by clbanjai** [Exawind/amr-wind#1113] [Q&A] [answered]
Hello, 

I am trying to run a simulation on AMR-Wind using the IEA Wind 15-Megawatt Offshore Wind Turbine. I am not sure what exactly is wrong, but this is the input file that I am currently working with 
[IEA-15MW_input_file.txt](https://github.com/user-attachments/files/16057071/IEA-15MW_input_file.txt)

The simulation runs for the first 0.2 seconds and aborts right after, and gives me the following error:
 
terminate called after throwing an instance of 'std::runtime_error'
  what():  FastIface: Error calling OpenFAST function:
FAST_Solution:FAST_AdvanceStates:AD_UpdateStates:BEMT_UpdateStates(node 50, blade 2):UA_UpdateStates:UA_UpdateDiscOtherState:ComputeKelvinChain:Mach number exceeds 1.0. Equations cannot be evaluated.

Has anyone encountered a similar error?

**Comment by tonyinme** [comment]
Hi @clbanjai , you are most likely seeing an instability from OpenFAST. What is the time-step in OpenFAST? Also, what is the structural dynamics model that you are using "CompElast" ?

**Reply by clbanjai** [reply]
The time-step for my amr-wind inp input file was 0.04 s (anything above this results in CFL>1). My fst file had a time-step of 0.005, which was the default value. I got the fst files from https://github.com/IEAWindTask37/IEA-15-240-RWT/tree/master/OpenFAST, and I am running IEA-15-240-RWT-Monopile.fst so the structural dynamics model I believe is IEA-15-240-RWT-Monopile_ElastoDyn.dat
I am new to using AMR-Wind and OpenFAST.

**Comment by tonyinme** [comment, marked as answer]
Ok, here are a few suggestions that you can try to improve your setup:

This should be set to 0:
https://github.com/IEAWindTask37/IEA-15-240-RWT/blob/2575756896ab45575ab1c094da76fde65a3d1d47/OpenFAST/IEA-15-240-RWT-Monopile/IEA-15-240-RWT-Monopile.fst#L14 

Make sure that this option is also set to 0
https://github.com/IEAWindTask37/IEA-15-240-RWT/blob/2575756896ab45575ab1c094da76fde65a3d1d47/OpenFAST/IEA-15-240-RWT-Monopile/IEA-15-240-RWT-Monopile_AeroDyn15.dat#L6 

Can you adjust the variable below to match the value in your aerodyn file?
"Actuator.TurbineFastLine.num_points_tower = 15"

https://github.com/IEAWindTask37/IEA-15-240-RWT/blob/2575756896ab45575ab1c094da76fde65a3d1d47/OpenFAST/IEA-15-240-RWT-Monopile/IEA-15-240-RWT-Monopile_AeroDyn15.dat#L115 

Also, what version of the codes (amr-wind and openfast) are you using?

**Comment by clbanjai** [comment]
I am using amr-wind version 2.1.0 and openfast 3.5.3.

Regarding https://github.com/IEAWindTask37/IEA-15-240-RWT/blob/2575756896ab45575ab1c094da76fde65a3d1d47/OpenFAST/IEA-15-240-RWT-Monopile/IEA-15-240-RWT-Monopile_AeroDyn15.dat#L6. I had it initially set to 1, which is the default value, but I getting the following error **Assertion `npts == (nrotor_pts + fi.num_pts_tower)'** failed, even though  "Actuator.TurbineFastLine.num_points_tower" and "Actuator.TurbineFastLine.num_points_blade" both matched the AeroDyn15 and ElastoDyn files, setting the wakemod to 2 fixed the issue. I tried setting it to 0 as suggest and I am getting the Assertion  error again. Perhaps there is something in my other files that prevents changing the wakemod.
